### ISSUE TITLE
when do we support cudnn4?

### ISSUE THREAD

**Issue by fayeshine** [tensorflow/tensorflow#714]
cudnn4 natively support batch normalization, when do we support cudnn4?
Actually, theano acts very fast, it support cudnn4 only several days after it comes out.

**Comment by zheng-xq** [comment]
@fayeshine, we will officially switch to Cudnn R4 when it is officially released. For now, it is a release candidate.

**Comment by zheng-xq** [comment]
After commit [22ebf0a](https://github.com/tensorflow/tensorflow/commit/22ebf0a94fd42af2d78b7964e836c92673ddfa31), you can build TensorFlow with Cudnn R3 and R4, after the following changes to your Cudnn library. 

For Cudnn R3:
$ ln -s libcudnn.so.7.0 libcudnn.so.6.5

For Cudnn R4:
$ ln -s libcudnn.so.4.0.4 libcudnn.so.6.5

This is currently still an unofficial feature. TensorFlow will officially switch to R4 soon after the Cudnn R4 final version is released.

The performance with R4 is better than R2. There are still a number of memory and performance improvements we are working on. Please stay tuned for more changes in this area. 

Thanks. 
-XQ

**Comment by fayeshine** [comment]
@zheng-xq ,thank you! hope to see the official cudnn4 comes out soon! I'm really interested to see how theano and tensowflow acts on cudnn4. I believe tensowflow will compile much faster and hope it can also run the training faster than theano.

**Comment by fayeshine** [comment]
@zheng-xq However, when do we start to support cuda 7.5 or upcoming 8.0?

**Comment by fayeshine** [comment]
@zheng-xq , cudnn v4 now is officially released

**Comment by zheng-xq** [comment]
We are working on upgrading to Cudnn R4, and test it on all platforms. We'll keep this thread posted.

**Comment by fayeshine** [comment]
@zheng-xq , and what about cuda 7.5?

**Comment by ppwwyyxx** [comment]
@fayeshine I think cuda 7.5 is also experimentally supported right now. See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#using-a-different-cuda-sdk-and-cudnn-versions
